.test-infra/jenkins/job_LoadTests_ParDo_Python.groovy

```diff
@@ -0,0 +1,153 @@
+                        iterations           : 10,
+                        number_of_counter_operations: 0,
+                        number_of_counters   : 1,
+                        num_workers          : 5,
+                        autoscaling_algorithm: 'NONE',
+                ]
+        ],
+        [
+                title        : 'ParDo Python Load test: 2GB 100 byte records 200 times',
+                itClass      : 'apache_beam.testing.load_tests.pardo_test:ParDoTest.testParDo',
+                runner       : CommonTestProperties.Runner.DATAFLOW,
+                jobProperties: [
+                        job_name             : 'load-tests-python-dataflow-batch-pardo-2-' + now,
+                        project              : 'apache-beam-testing',
+                        temp_location        : 'gs://temp-storage-for-perf-tests/smoketests',
+                        publish_to_big_query : true,
+                        metrics_dataset      : datasetName,
+                        metrics_table        : 'python_dataflow_batch_pardo_2',
+                        input_options        : '\'{' +
+                                '"num_records": 20000000,' +
+                                '"key_size": 10,' +
+                                '"value_size": 90}\'',
+                        iterations           : 200,
+                        number_of_counter_operations: 0,
+                        number_of_counters   : 1,
+                        num_workers          : 5,
+                        autoscaling_algorithm: 'NONE',
+                ]
+        ],
+        [
+                title        : 'ParDo Python Load test: 2GB 100 byte records 10 counters',
+                itClass      : 'apache_beam.testing.load_tests.pardo_test:ParDoTest.testParDo',
+                runner       : CommonTestProperties.Runner.DATAFLOW,
+                jobProperties: [
+                        job_name             : 'load-tests-python-dataflow-batch-pardo-3-' + now,
+                        project              : 'apache-beam-testing',
+                        temp_location        : 'gs://temp-storage-for-perf-tests/smoketests',
+                        publish_to_big_query : true,
+                        metrics_dataset      : datasetName,
+                        metrics_table        : 'python_dataflow_batch_pardo_3',
+                        input_options        : '\'{' +
+                                '"num_records": 20000000,' +
+                                '"key_size": 10,' +
+                                '"value_size": 90}\'',
+                        iterations           : 10,
+                        number_of_counter_operations: 10,
+                        number_of_counters   : 1,
+                        num_workers          : 5,
+                        autoscaling_algorithm: 'NONE',
+                ]
+        ],
+]}
+
+def loadTestConfigurationManyCounters = { datasetName -> [
```

Comment:
I don't think splitting the job to 2 jobs is the solution here. It is very weird that it takes so much time (above 2 hours) to run, whereas java needs 28 minutes to run all the tests (as we discussed offline). Could you investigate this a little bit more? Maybe the pipeline shape is not as we think it is?

Reply:
I have double-checked the code and it seems fine. The problem is poor performance of metrics operations in Python — test case with 10 iterations and 100 counter operations needed almost 3 hours to complete.

The solution is to lower the number of iterations to 1, which makes the test way faster. Job split will be also unnecessary.